cli/src/complete.rs

```diff
@@ -271,6 +271,31 @@ pub fn operations() -> Vec<CompletionCandidate> {
     })
 }
 
+pub fn workspaces() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj, _| {
+        let output = jj
+            .arg("--config-toml")
+            .arg("templates.commit_summary = 'description.first_line()'")
+            .arg("workspace")
+            .arg("list")
+            .output()
+            .map_err(user_error)?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
+
+        Ok(stdout
+            .lines()
+            .map(|line| {
+                let (name, desc) = line.split_once(": ").expect("output should contain ': '");
+                let help = match desc {
+                    "" => Some("(no description set)".to_string().into()),
```

Comment:
nit: this could be moved to template `if(description, description.first_line(), description_placeholder)` for consistency, though the result is mostly identical.